climada/hazard/centroids/centr.py

```diff
@@ -820,169 +351,90 @@ def get_closest_point(self, x_lon, y_lat, scheduler=None):
         idx_close : int
             Index of centroid in internal ordering of centroids.
         """
-        if self.meta:
-            if not self.lat.size or not self.lon.size:
-                self.set_meta_to_lat_lon()
-            i_lat, i_lon = rasterio.transform.rowcol(self.meta['transform'], x_lon, y_lat)
-            i_lat = np.clip(i_lat, 0, self.meta['height'] - 1)
-            i_lon = np.clip(i_lon, 0, self.meta['width'] - 1)
-            close_idx = int(i_lat * self.meta['width'] + i_lon)
-        else:
-            self.set_geometry_points(scheduler)
-            close_idx = self.geometry.distance(Point(x_lon, y_lat)).values.argmin()
+        close_idx = self.geometry.distance(Point(x_lon, y_lat)).values.argmin()
         return self.lon[close_idx], self.lat[close_idx], close_idx
 
-    def set_region_id(self, scheduler=None):
+    def set_region_id(self, level='country', overwrite=False):
```

Comment:
I think this is needed for user-friendliness. You do not want to have to recreate new centroids all the time when you change the region_id. Also, if I remember correctly this is needed for the way Litpop is implemented.

The question is whether the name `region_id` should be reserved for `country_id` or it should also allow other types of regional information such as admin1.

Reply:
Because it is quite expensive both in memory and cpu if you have the whole world at 1x1km resolution for instance.

Reply:
That's mainly because of the `gpd.points_from_xy`, right? Because just copying an array that contains the whole world in 1x1km resolution several times shouldn't be an issue.

Reply:
I can't imagine a scenario where this is repeated more than two or three times without stuff being garbage collected.

Reply:
Good point.